### PR TITLE
Adding spotless

### PR DESCRIPTION
**Pull request by arrekb**
Everything formatted except for the `package-name` (see `"disabled_rules" to "package-name"` in `build.gradle.kts`).
I'm not sure if you want a difference between the package name and the repository name.
If so, I will gladly do it.

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -84,3 +85,21 @@ publishing {
         }
     }
 }
+
+configure<com.diffplug.gradle.spotless.SpotlessExtension> {
+    kotlin {
+        ktlint().editorConfigOverride(
+            mapOf(
+                "indent_size" to "4",
```
Comment (platan) [inline build.gradle.kts:93]: Why do you set `indent_size`?

File: build.gradle.kts [hunk 2]
```diff
@@ -84,3 +85,21 @@ publishing {
...
+                "disabled_rules" to "package-name"
+            )
+        )
+    }
+    kotlinGradle {
+        target("*.gradle.kts")
+        ktlint().editorConfigOverride(
+            mapOf("indent_size" to "4")
```
Comment (platan) [inline build.gradle.kts:101]: Why do you set `indent_size`?

File: build.gradle.kts [hunk 3]
```diff
@@ -84,3 +85,21 @@ publishing {
         }
     }
 }
+
+configure<com.diffplug.gradle.spotless.SpotlessExtension> {
+    kotlin {
+        ktlint().editorConfigOverride(
+            mapOf(
+                "indent_size" to "4",
+                "disabled_rules" to "package-name"
```
Comment (platan) [inline build.gradle.kts:94]: When I run `spotlessApply` `Property 'disabled_rules' is deprecated: Rename property 'disabled_rules' to 'ktlint_disabled_rules' in all '.editorconfig' files.` is printed 900 times:
```sh
➜  tests-gantt-chart git:(adding_spotless) ./gradlew spotlessApply --rerun-tasks | grep "is deprecated" -m 1
Property 'disabled_rules' is deprecated: Rename property 'disabled_rules' to 'ktlint_disabled_rules' in all '.editorconfig' files.
➜  tests-gantt-chart git:(adding_spotless) ./gradlew spotlessApply --rerun-tasks | grep "is deprecated" -c
900
```

File: build.gradle.kts [hunk 4]
```diff
@@ -84,3 +85,21 @@ publishing {
...
+            )
+        )
+    }
+    kotlinGradle {
+        target("*.gradle.kts")
+        ktlint().editorConfigOverride(
+            mapOf("indent_size" to "4")
+        )
+    }
+    isEnforceCheck = false
```
Comment (platan) [inline build.gradle.kts:104]: I would be good to enable spotless:
- apply spotless for every commit (git hook) for every user by default
- check formatting on CI
Would you mind to add such features?

**Review by platan — CHANGES_REQUESTED** [review]
I've left some comments. I think we should merge other open PRs before merging this one (I don't want to make git conflicts).

File: build.gradle.kts (hunk 1 above)
Comment (arrekb) [reply build.gradle.kts]: Removed for `kotlin`

File: build.gradle.kts (hunk 2 above)
Comment (arrekb) [reply build.gradle.kts:100]: I found spotless does not fix intendations in build.gradle without `set indent_size`

File: build.gradle.kts (hunk 3 above)
Comment (arrekb) [reply build.gradle.kts:96]: Fixed, more info in comment in `build.gradle.kts`

File: build.gradle.kts (hunk 4 above)
Comment (arrekb) [reply build.gradle.kts]: AFAIK there is not a way to force hooks upon clone so I prepared `src/etc/githook/commit-msg` instead
Added step `Check with Gradle` to `ci.yml`

File: build.gradle.kts (hunk 2 above)
Comment (platan) [reply build.gradle.kts:103]: Thank you for explanation.

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -19,6 +19,10 @@ jobs:
       with:
         java-version: '11'
         distribution: 'temurin'
+    - name: Check with Gradle
+      uses: gradle/gradle-build-action@67421db6bd0bf253fb4bd25b31ebb98943c375e1
+      with:
+        arguments: check
```
Comment (platan) [inline .github/workflows/ci.yml:25]: Why did you add `check`? The next step is `build`, which run spotless as well. In my opinion we do not need `check`.

File: src/etc/githook/commit-msg [hunk 6]
```diff
@@ -0,0 +1,5 @@
+#!/bin/sh
+
+# To apply spotless for every commit copy this file to folder .git/hooks/
+# or add below line to existing .git/hooks/commit-msg
```
Comment (platan) [inline src/etc/githook/commit-msg:4]: In my opinion `.git/hooks/commit-msg` is not a good place for this check. Lets look at samples:
```
cat .git/hooks/commit-msg.sample | head -5
#!/bin/sh
#
# An example hook script to check the commit log message.
# Called by "git commit" with one argument, the name of the file
# that has the commit message.  The hook should exit with non-zero
```

`pre-commit` seeds to be more suitable:
```
cat .git/hooks/pre-commit.sample | head -5
#!/bin/sh
#
# An example hook script to verify what is about to be committed.
# Called by "git commit" with no arguments.  The hook should
# exit with non-zero status after issuing an appropriate message if
```

**Review by platan — CHANGES_REQUESTED** [review]

File: build.gradle.kts (hunk 4 above)
Comment (platan) [reply build.gradle.kts]: In the js/npm world there are https://www.npmjs.com/package/simple-git-hooks and https://www.npmjs.com/package/husky, which configure git hooks. 

I found https://github.com/jakemarsden/git-hooks-gradle-plugin. I think we can use it. I tested it and it works properly. Projects it not very popular, but has small codebase and has no dependencies.

Comment (platan) [reply build.gradle.kts]: Btw, you added `spotlessCheck` to git hook. In `apply spotless for every commit (git hook) for every user by default` I thought about `spotlessApply`. But we do not have to add this in this PR. Using `spotlessApply` in git hook is more complicated. because it should add modified (fixed) filed to git index. Moreover it would be good apply spotless only to changed files.

Comment (arrekb) [reply build.gradle.kts]: Added [plugin](https://github.com/jakemarsden/git-hooks-gradle-plugin) and configured pre-commit hook

File: .github/workflows/ci.yml (hunk 5 above)
Comment (arrekb) [reply .github/workflows/ci.yml]: Removed

File: build.gradle.kts [hunk 7]
```diff
@@ -84,3 +86,27 @@ publishing {
...
+        target("*.gradle.kts")
+        ktlint().editorConfigOverride(
+            mapOf("indent_size" to "4")
+        )
+    }
+}
+
+gitHooks {
+    setHooks(mapOf("pre-commit" to "spotlessCheck"))
+    setHooksDirectory(layout.projectDirectory.dir("/tmp/tests-execution-chart/.git/hooks"))
```
Comment (platan) [inline build.gradle.kts:111]: Setting hooks directory to `/tmp/tests-execution-chart/.git/hooks` will not result with working git hook. We should not set `setHooksDirectory`. By default hooks will be created in .git/hooks and it is ok.

Comment (arrekb) [reply build.gradle.kts]: It seemed to me working. I tested with the default location first, then changed it but not removed the first hook :-(
setHooksDirectory removed

**Review by platan — APPROVED** [review]
Thank you for this change.
